src/Raven.Server/Documents/Queries/Results/TimeSeries/AverageAggregation.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Raven.Client.Documents.Queries.TimeSeries;
+using Raven.Server.Documents.TimeSeries;
+
+namespace Raven.Server.Documents.Queries.Results.TimeSeries
+{
+    public class AverageAggregation : TimeSeriesAggregationBase, ITimeSeriesAggregation
+    {
+        private readonly List<double> _count;
+
+        public AverageAggregation(string name = null) : base(AggregationType.Average, name)
+        {
+            _count = new List<double>();
+        }
+
+        void ITimeSeriesAggregation.Segment(Span<StatefulTimestampValue> values, bool isRaw)
+        {
+            if (isRaw == false)
+            {
+                Debug.Assert(values.Length % 6 == 0);
+                var originalNumOfValues = values.Length / 6;
+
+                InitValuesIfNeeded(originalNumOfValues);
+
+                for (int i = 0; i < originalNumOfValues; i++)
+                {
+                    var index = i * 6;
+
+                    var val = values[index + (int)AggregationType.Sum];
+                    _values[i] += val.Sum;
+                    
+                    val = values[index + (int)AggregationType.Count];
+                    _count[i] += (long)val.Sum;
+                }
+
+                return;
+            }
+
+            InitValuesIfNeeded(values.Length);
+
+            for (int i = 0; i < values.Length; i++)
+            {
+                var val = values[i];
+
+                _values[i] += val.Sum;
+                _count[i] += val.Count;
+            }
+        }
+
+        void ITimeSeriesAggregation.Step(Span<double> values, bool isRaw)
+        {
+            if (isRaw == false)
+            {
+                Debug.Assert(values.Length % 6 == 0);
+                var originalNumOfValues = values.Length / 6;
+
+                InitValuesIfNeeded(originalNumOfValues);
+
+                for (int i = 0; i < originalNumOfValues; i++)
+                {
+                    var index = i * 6;
+
+                    var val = values[index + (int)AggregationType.Sum];
+                    _values[i] += val;
+
+                    val = (long)values[index + (int)AggregationType.Count];
+                    _count[i] += val;
+                }
+                return;
+            }
+
+            InitValuesIfNeeded(values.Length);
+
+            for (int i = 0; i < values.Length; i++)
+            {
+                _values[i] += values[i];
+                _count[i]++;
+            }
+        }
+
+        IEnumerable<double> ITimeSeriesAggregation.GetFinalValues(DateTime? @from, DateTime? to, double? scale = null)
+        {
```

Comment:
Can you check the warning?
```
src/Raven.Server/Documents/Queries/Results/TimeSeries/AverageAggregation.cs#L83
The default value specified for parameter 'scale' will have no effect because it applies to a member that is used in contexts that do not allow optional arguments
```

Probably just remove `= null` from `scale` here. It is an explicit interface impl.

Reply:
done